File: Solution Using C#/Fawry.Ecommerce/ExpirableProduct.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Fawry.Ecommerce
{
    internal class ExpirableProduct:Product
    {
        public DateTime ExpirationDate { get; set; }
        public ExpirableProduct(string name, double price, int quantity, DateTime expirationDate)
            : base(name, price, quantity)
        {
            ExpirationDate = expirationDate;
        }
        public override bool IsExpired()
        {
            return DateTime.Now > ExpirationDate;
        }

        public override string ToString()
        {
            return $"{Name} - {Price:C} - {Quantity} in stock - Expires on {ExpirationDate.ToShortDateString()}";
        }
    }
}

File: Solution Using Java/src/models/ExpirableProduct.java
package models;

import java.time.LocalDate;

public class ExpirableProduct extends  Product{
    private LocalDate expirationDate;
    public ExpirableProduct(String name, double price,int quantity,LocalDate expirationDate)
    {
        super(name,price,quantity);
        this.expirationDate = expirationDate;
    }
    public LocalDate getExpirationDate() {
        return expirationDate;
    }
    public void setExpirationDate(LocalDate expirationDate) {
        this.expirationDate = expirationDate;
    }

    @Override
    public boolean isExpired() {
        return LocalDate.now().isAfter(expirationDate);
    }

    @Override
    public String toString() {
        return getName() + " - $"+ String.format("%.2f",getPrice()) + getQuantity() +" in stock - Expires on "+getExpirationDate();
    }
}
